### DISCUSSION TITLE
VoxelDataStore and VoxelColorStore why both?

### DISCUSSION THREAD

**Discussion by TheAtomAnt** [Eldemarkki/Marching-Cubes-Terrain#56] [Q&A]
Hi there,

I'm wondering about a design decision to separate VoxelDataStore  and VoxelColorStore, and the corresponding voxel data. Ultimately it's an array of bytes, and an array of Color32. But why split that data into two parts?

**Comment by Eldemarkki** [comment]
That is fixed in the develop branch. I created a PerChunkStore<T> and a PerVoxelStore<T>. The PerVoxelStore is just a PerChunkStore<NativeArray<T>>. 

Here are the links to those files
- https://github.com/Eldemarkki/Marching-Cubes-Terrain/blob/develop/Assets/VoxelTerrain/Core/VoxelData/PerChunkStore.cs
- https://github.com/Eldemarkki/Marching-Cubes-Terrain/blob/develop/Assets/VoxelTerrain/Core/VoxelData/PerVoxelStore.cs

I use the PerChunkStore for storing the chunk properties, and I use PerVoxelStore for storing the voxel data (bytes) in PerVoxelStore<byte> and colors in PerVoxelStore<Color32>. They still require different files (e.g. VoxelDataStore.cs and VoxelColorStore.cs) because they need some overrides.
